### DISCUSSION TITLE
get() after props.name

### DISCUSSION THREAD

**Discussion by ghost** [framesurge/perseus#217] [General]
In [examples/website/app_in_a_file/src/main.rs](https://github.com/framesurge/perseus/blob/main/examples/website/app_in_a_file/src/main.rs) there's get() after props.name. After I replace these code 

> view! { cx,
>         h1 { (format!(
>             "Hello, {}!",
>             props.name.get()
>         )) }
>         input(
>             placeholder = "Name",
>             bind:value = props.name
>         )

by 

> view! { cx,
>         h1 { (format!(
>             "Hello, {}!",
>             props.name
>         )) }
>         input(
>             placeholder = "Name",
>             bind:value = props.name
>         )

, it still compiles. What does get() do?

**Comment by lukechu10** [comment]
> In [examples/website/app_in_a_file/src/main.rs](https://github.com/framesurge/perseus/blob/main/examples/website/app_in_a_file/src/main.rs) there's get() after props.name. After I replace these code
> 
> > view! { cx,
> > h1 { (format!(
> > "Hello, {}!",
> > props.name.get()
> > )) }
> > input(
> > placeholder = "Name",
> > bind:value = props.name
> > )
> 
> by
> 
> > view! { cx,
> > h1 { (format!(
> > "Hello, {}!",
> > props.name
> > )) }
> > input(
> > placeholder = "Name",
> > bind:value = props.name
> > )
> 
> , it still compiles. What does get() do?

This is because `Signal` implements `Display` so it calls .get() automatically when you format it. However, in other cases, you still need to explicitly call .get()
